src/main/java/seedu/address/commons/core/index/Index.java

```diff
@@ -58,12 +76,19 @@ public boolean equals(Object other) {
             return false;
         }
 
+        if (other == Index.WILDCARD) {
+            return this == other;
+        }
+
         Index otherIndex = (Index) other;
         return zeroBasedIndex == otherIndex.zeroBasedIndex;
     }
 
     @Override
     public String toString() {
+        if (this.equals(Index.WILDCARD)) {
```

Comment:
If there is only 1 instance of wildcard, maybe we want to use == instead since it is clearer and simpler also.

Reply:
valid